src/shapes/ActiveSelection.ts

```diff
@@ -75,6 +75,13 @@ export class ActiveSelection extends Group {
       const parent = object.group;
       parent._exitGroup(object);
       object.__owningGroup = parent;
+      if (
+        this._objects.includes(parent) &&
+        parent._objects.every((object) => this._objects.includes(object))
+      ) {
+        // in case group and all it's objects are selected we deselect group
+        this.remove(parent);
+      }
```

Comment:
this kind of interactivity with groups opens for a lot of edge cases.
I don't think automatically deselecting things outside user clicks is correct since we are assuming which is the correct UX here.
Why is deselecting the parent or the group or anything correct over the other option?
Does it create an interaction issue or just feel strange?

Reply:
with interactive groups it will be possible to deselect them in a multi selection interaction if only one object remains unselected in group since the object becomes the deselection target and the group will hang around, **SELECTED**

Reply:
I agree maybe it should be moved to multiSelectAdd

Reply:
I remember you said groups are simply containers.
What meaning does an empty selected container have in a selection context?

Reply:
Groups were simple containers, yes, meaning the would just contain object and render them.
Now they are interactive, and the surface edge cases.
Most of them should be left to the developer, and we should patch the ones that can be caused by the user alone.
Probably is enough to avoid rendering the control boxes, i m not sure at this point since it doesn't make sense to select a group and the objects inside.

What i would answer quickly would be that if the group is selected, you can't select objects inside.
And probably is the solution i would adapt if it  simplify our life.

Reply:
> What i would answer quickly would be that if the group is selected, you can't select objects inside.
> And probably is the solution i would adapt if it simplifies our life.

Not bad

Reply:
Your suggestion makes sense. If we agree on it I can try a patch